float64 theta
float64 P_theta
float64 z